marlowe-runtime/cip30-demo/Main.hs

```diff
@@ -21,9 +37,22 @@ newtype WalletName = WalletName String
     , Blaze.ToMarkup
     )
 
+newtype PostWalletCreateRequestDTO = PostWalletCreateRequestDTO { source :: Text }
+  deriving Show
+
+data PostWalletCreateResponseDTO = PostWalletCreateResponseDTO
+  { unsignedTransaction :: Text
+  , contractId :: Text
+  }
+
+instance Aeson.FromJSON PostWalletCreateRequestDTO where
+  parseJSON = Aeson.withObject "PostWalletCreateRequestDTO" \o ->
+    PostWalletCreateRequestDTO <$> o .: "source"
```

Comment:
If you derive `Generic` for `PostWalletCreateRequestDTO` you can omit this `parseJSON` definition.

Reply:
Right, it's just a temporary thing. I will later add more fields to `PostWalletCreateRequestDTO` and then it will be derived correctly. I don't think it's treated as a record with only one field which means that I will need to do some rework in the inlined JavaScipt if I derive it now and I'd rather have the compiler help me later.